src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -545,50 +526,67 @@ public static bool TryParseExact([NotNullWhen(true)] string? input, [NotNullWhen
 
-        public bool TryFormat(Span<char> destination, out int charsWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null) =>
-            TimeSpanFormat.TryFormat(this, destination, out charsWritten, format, formatProvider);
+        public bool TryFormat(Span<char> destination, out int charsWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null)
+            => TimeSpanFormat.TryFormat(this, destination, out charsWritten, format, formatProvider);
 
         /// <inheritdoc cref="IUtf8SpanFormattable.TryFormat" />
-        public bool TryFormat(Span<byte> utf8Destination, out int bytesWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null) =>
-            TimeSpanFormat.TryFormat(this, utf8Destination, out bytesWritten, format, formatProvider);
+        public bool TryFormat(Span<byte> utf8Destination, out int bytesWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null)
+            => TimeSpanFormat.TryFormat(this, utf8Destination, out bytesWritten, format, formatProvider);
 
         #endregion
 
         public static TimeSpan operator -(TimeSpan t)
         {
-            if (t._ticks == MinValue._ticks)
-                throw new OverflowException(SR.Overflow_NegateTwosCompNum);
+            if (t._ticks == MinTicks)
+            {
+                ThrowHelper.ThrowOverflowException_NegateTwosCompNum();
+            }
             return new TimeSpan(-t._ticks);
         }
 
-        public static TimeSpan operator -(TimeSpan t1, TimeSpan t2) => t1.Subtract(t2);
+        public static TimeSpan operator -(TimeSpan t1, TimeSpan t2)
+        {
+            long result = t1._ticks - t2._ticks;
+            long t1Sign = t1._ticks >> 63;
+
+            if ((t1Sign != (t2._ticks >> 63)) && (t1Sign != (result >> 63)))
+            {
+                // Overflow if signs of operands was different and result's sign was opposite.
+                // >> 63 gives the sign bit (either 64 1's or 64 0's).
+                ThrowHelper.ThrowOverflowException_TimeSpanTooLong();
+            }
+            return new TimeSpan(result);
+        }
 
         public static TimeSpan operator +(TimeSpan t) => t;
 
-        public static TimeSpan operator +(TimeSpan t1, TimeSpan t2) => t1.Add(t2);
+        public static TimeSpan operator +(TimeSpan t1, TimeSpan t2)
+        {
+            long result = t1._ticks + t2._ticks;
+            long t1Sign = t1._ticks >> 63;
+
+            if ((t1Sign == (t2._ticks >> 63)) && (t1Sign != (result >> 63)))
+            {
+                // Overflow if signs of operands was identical and result's sign was opposite.
```

Comment:
just curious, you could have used >>> instead of >> here, right? then it would only be the low order bit holding the sign. does it make any difference to perf?

Reply:
Signed and unsigned shift have the same cost in x64 and Arm64 hardware, so no real difference